### PR TITLE
new(tests): EIP-3540: execution semantics changes

### PR DESCRIPTION
**Pull request by shemnon**
## 🗒️ Description

Test the execution semantics changes around EXTCODE* and DELEGATECALL

Covers the tests in https://github.com/ethereum/tests/blob/9c84e0db14adf4be4f93640e51f6953d583d8c49/src/EIPTestsFiller/StateTests/stEOF/stEIP3540/EOF1_CallsFiller.yml and the EXTCODE* tests in https://github.com/ethereum/tests/blob/9c84e0db14adf4be4f93640e51f6953d583d8c49/src/EIPTestsFiller/StateTests/stEOF/stEIP3540/EOF1_ExecutionFiller.yml

## ✅ Checklist
- [ ] All: Set appropriate labels for the changes.
- [ ] All: Considered squashing commits to improve commit history.
- [ ] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [ ] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Comment by shemnon** [comment]
This build fails because it is using Geth to attempt to fill the tests, which is problematic because Geth doesn't have the features implemented that these tests are being written for.  

Is gating on Geth's implementation for merging really a good idea?

File: tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_extcode.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    legacy_code = Op.PUSH1(2) + Op.JUMPDEST + Op.STOP
+    pre = {
+        TestAddress: Account(
+            balance=1000000000000000000000,
+            nonce=1,
+        ),
+        address_eof_contract: Account(
+            code=Container(
+                sections=[
+                    Section.Code(code=Op.RJUMP[0] + Op.STOP, code_output=NON_RETURNING_SECTION)
```
Comment (marioevz) [inline tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_extcode.py:42]:
```suggestion
                    Section.Code(code=Op.RJUMP[0] + Op.STOP, code_outputs=NON_RETURNING_SECTION)
```
It took me 20 minutes to find out this typo was the reason why we were getting `INVALID_FIRST_SECTION_TYPE` in this test.

I think we might need to add the proper keyword arguments (with defaults) to `Section.Code`.

**Review by marioevz — COMMENTED** [review]
Thanks for the tests, some comments.

File: tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_calls.py [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    env = Environment()
+    calling_contract_address = Address(0x1000000)
+    destination_contract_address = Address(0x1000001)
+
+    caller_contract = Container(
+        sections=[
+            Section.Code(
+                code=Op.SSTORE(slot_call_result, opcode(destination_contract_address, *suffix))
+                + Op.SSTORE(slot_returndata, Op.RETURNDATALOAD(0))
+                + Op.SSTORE(slot_code_worked, value_code_worked)
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_calls.py:481]: The spec for this just changed - https://github.com/ipsilon/eof/pull/90 - Impls are out of date.

File: tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_extcode.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+)
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.eof.v1.constants import NON_RETURNING_SECTION
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-4750.md"
+REFERENCE_SPEC_VERSION = "90f716078d0b08ce508a1e57803f885cc2f2e15e"
+
+
+def test_legacy_calls_eof_sstore(
```
[the remark this answers is not in the document]
Comment (shemnon) [reply tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_extcode.py:23]: This is unclear, who is the caller (legacy, EOF mid-creation) and who is the receiver?  Legacy, other EOF?  Both?

[the remark this answers is not in the document]
Comment (marioevz) [reply tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_extcode.py:28]: I was referring to an EOF initcode doing EXTCALL to a legacy contract, and then this legacy contract does `EXTCODESIZE`, `EXTCODEHASH`, `EXTCODECOPY` to the address where the EOF initcode is running.

[the remark this answers is not in the document]
Comment (marioevz) [reply tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_extcode.py:28]: I will open up a follow up PR to test this case.

**Review by marioevz — APPROVED** [review]
Thanks!

**Pull request merged**